### PR TITLE
Improve SOLID in NewMonthUCI

### LINKED ISSUE #47: Improve SOLID in NewMonthUCI

### PR DESCRIPTION
**Pull request by YinLingithub**
Made implementation changes to addMonth() in SessionStorage to move responsibility of check-if-month-already-exist-in-session from the NewMonthUCI to SessionStorage. 
Implemented new private method createNewMonth() to create new MonthlyStorage object in NewMonthUCI. 

Made method signature changes in relevant test cases.

### CODE REVIEW

File: src/main/java/use_cases/create_new_month/NewMonthUCI.java [hunk 1]
```diff
@@ -21,49 +21,50 @@ public NewMonthUCI(NewMonthOB outputBoundary) {
...
-            }
+    /**
+     * Create a new MonthlyStorage object.
+     * @param session the SessionStorage to store the new MonthlyStorage
+     * @param monthID the monthID of the new MonthlyStorage
+     * @param budgetValue the budget for the new MonthlyStorage
+     * @return new MonthlyStorage object
+     * @throws EntityException if an error in the program causes two Expenese in recurData to have the same name
+     */
+    private MonthlyStorage createNewMonth(SessionStorage session, int monthID, double budgetValue) throws EntityException {
```
Comment (katarinaavucic) [inline src/main/java/use_cases/create_new_month/NewMonthUCI.java:61]: Good idea! I like how you extracted this code into a new method.

File: src/test/java/use_cases/create_new_month/NewMonthUCITest.java [hunk 2]
```diff
@@ -18,8 +18,11 @@ class NewMonthUCITest {
     static MonthlyStorage monthlyStorage1;
     static SessionStorage sessionStorage;
 
+    /**
+     * @throws EntityException if an error occur with addMonth() in SessionStorage
```
Comment (katarinaavucic) [inline src/test/java/use_cases/create_new_month/NewMonthUCITest.java:23]: You are missing a comment explaining what the method does!

File: src/test/java/use_cases/main_menu/SessionSaveUCITest.java [hunk 3]
```diff
@@ -16,8 +17,11 @@
 class SessionSaveUCITest {
     static SessionStorage sampleSession;
 
+    /**
+     * @throws EntityException if an error occur with addMonth() in SessionStorage
```
Comment (katarinaavucic) [inline src/test/java/use_cases/main_menu/SessionSaveUCITest.java:22]: Missing one here as well!

File: src/test/java/use_cases/monthly_menu/UpdateViewUCITest.java [hunk 4]
```diff
@@ -22,8 +22,11 @@ class UpdateViewUCITest {
     static MonthlyStorage monthlyStorage;
     static SessionStorage sessionStorage;
 
+    /**
+     * @throws EntityException if an error occur with addMonth() in SessionStorage
```
Comment (katarinaavucic) [inline src/test/java/use_cases/monthly_menu/UpdateViewUCITest.java:27]: And here!

File: src/test/java/use_cases/session_load/SessionLoadUCITest.java [hunk 5]
```diff
@@ -14,8 +15,11 @@
 class SessionLoadUCITest {
     static SessionStorage sampleSession;
 
+    /**
+     * @throws EntityException if an error occur with addMonth() in SessionStorage
```
Comment (katarinaavucic) [inline src/test/java/use_cases/session_load/SessionLoadUCITest.java:20]: Also here :).

**Review by katarinaavucic — CHANGES_REQUESTED** [review]
Looks good! You are just missing some Javadocs.

File: src/test/java/use_cases/create_new_month/NewMonthUCITest.java (hunk 2 above)
Comment (YinLingithub) [reply src/test/java/use_cases/create_new_month/NewMonthUCITest.java:23]: Updated!

File: src/test/java/use_cases/main_menu/SessionSaveUCITest.java (hunk 3 above)
Comment (YinLingithub) [reply src/test/java/use_cases/main_menu/SessionSaveUCITest.java:22]: Updated!

File: src/test/java/use_cases/monthly_menu/UpdateViewUCITest.java (hunk 4 above)
Comment (YinLingithub) [reply src/test/java/use_cases/monthly_menu/UpdateViewUCITest.java:27]: Updated!

File: src/test/java/use_cases/session_load/SessionLoadUCITest.java (hunk 5 above)
Comment (YinLingithub) [reply src/test/java/use_cases/session_load/SessionLoadUCITest.java:20]: Updated!

**Review by katarinaavucic — APPROVED** [review]
Looks great! Nice fix to better adhere to SOLID.

**Review by aricasas — APPROVED** [review]
Great job! The createOutput method in the UCI looks way cleaner and it's easier to follow. Smart choice for the helper method. 

The new tests and documentation on the tests seems good to me.

**Review by luluwuwu — APPROVED** [review]
Looking good!

**Pull request merged**